### PR TITLE
add suggestion ..Default::default() for remaining struct fields in a constructor

### LINKED ISSUE #6492: Suggestion: `..Default::default()` for remaining struct fields in a constructor
IntelliSense to suggest `..Default::default()` when after the last comma of a struct constructor if it has public fields and a `Default` impl.

Currently I just use a snippet in VS Code:
```javascript
	"..Default::default()": {
		"prefix": [
			"..Default::default()"
		],
		"body": [
			"..Default::default()"
		],
		"description": "Fill remaining fields with default values"
	}
```

image

### PR DESCRIPTION
**Pull request by bnjjj**
I'm not sure I should import `assists` crate inside `completions`, maybe we should move out `FamousDefs` from `assists` ? Let me know :) 

close #6492

Signed-off-by: Benjamin Coenen <5719034+bnjjj@users.noreply.github.com>

### CODE REVIEW

**Comment by SomeoneToIgnore** [comment]
IMO we need some utils shared module between assists and completions indeed: `insert_use` is another example of things that could be reused in both modules.

**Comment by bnjjj** [comment]
Maybe a crate called `codegen_utils` ?

**Comment by SomeoneToIgnore** [comment]
I've just checked my code for auto imports for usages from assists: `use assists::utils::{insert_use, mod_path_to_ast, ImportScope};`

Not sure if all of them are 100% related to `codegen_utils`, but have no better idea for a name now.

**Comment by flodiebold** [comment]
Do we want to automatically insert the use though? We could instead just insert a qualified path to `Default`, and let the user decide whether to auto-import it.

File: crates/assists/src/utils.rs [hunk 1]
```diff
@@ -257,6 +257,12 @@ pub mod convert {
     }
 }
 
+pub mod default {
+    pub trait Default: Sized {
```
Comment (flodiebold) [inline crates/assists/src/utils.rs:261]: `Sized` isn't defined in the fixture, so it's probably better to leave it out

**Review by flodiebold — APPROVED** [review]

**Comment by flodiebold** [comment]
bors d+

**Comment by bnjjj** [comment]
@flodiebold should I move `FamousDefs` to another crate or just keep it like this ?

**Comment by SomeoneToIgnore** [comment]
To elaborate: I'm not insisting to create the crate now, we can merge the PR and design the crate in another one.

**Pull request merged**